tests/test_plot_manager.py

```diff
@@ -0,0 +1,177 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from .common import test_result_collector as trc
+from .common.test_utils import convert_to_bytes, ROOT_DIR
+from .mocks.mock_config import MockConfig
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_analyze \
+    import ConfigCommandAnalyze
+
+from model_analyzer.plots.plot_manager import PlotManager
+from model_analyzer.result.result_manager import ResultManager
+from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
+
+from filecmp import cmp
+from shutil import rmtree
+from unittest.mock import MagicMock, patch
+
+import pickle
+
+
+class TestPlotManager(trc.TestResultCollector):
+
+    def setUp(self):
+        self._create_single_model_result_manager()
+        self._create_multi_model_result_manager()
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_single_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/single-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._single_model_config,
+            result_manager=self._single_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        # file = open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.pkl", "wb")
+        # pickle.dump(plot_manager._simple_plots, file)
+        # file.close()
+
+        file = open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.pkl", "rb")
+        golden_plot_manager_simple_plots = pickle.load(file)
+        file.close()
+
+        golden = golden_plot_manager_simple_plots['add_sub'][
+            'throughput_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager_simple_plots['add_sub'][
+            'gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def test_multi_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/multi-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._multi_model_config,
+            result_manager=self._multi_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        # file = open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.pkl", "wb")
+        # pickle.dump(plot_manager._simple_plots, file)
+        # file.close()
+
+        file = open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.pkl", "rb")
```

Comment:
It's more pythonic to use [context managers](https://book.pythontips.com/en/latest/context_managers.html#context-managers) for this:
```
with open(filename, "rb"):
  golden_plot_manager_simple_plots = ...
```
# file automatically closed outside of context manager scope